src/scalarstats.jl

```diff
@@ -438,7 +438,17 @@ zscore(X::AbstractArray{<:Real}, dim::Int) = ((μ, σ) = mean_and_std(X, dim); z
 Compute the entropy of a collection of probabilities `p`, optionally specifying a real number
 `b` such that the entropy is scaled by `1/log(b)`.
 """
-entropy(p) = -sum(pi -> pi * log(pi), p)
+function entropy(p)
```

Comment:
I'd forgotten that `xlogx` isn't available here. We should probably move it here togeher with things like `logistic`. 

Probably better to just mimic `xlogx` with something like `-sum(pᵢ -> iszero(pᵢ) ? zero(pᵢ) : pᵢ * log(pᵢ), p)` since it will throw for negative probabilities.

Reply:
`xlogx` sounds so general that it could live in Base, no? We're trying to move things to Statistics so it could go there too. Anyway it doesn't make a lot of sense in StatsFuns.

Reply:
Why can't you just call `StatsFuns.xlogx`?

Reply:
Importing StatsFuns (which depends on R) just for xlogx seemed overkill

Reply:
Ah I thought it was already imported. Yes, we could definitely add it here.